src/Maestro/maestro-angular/src/app/model/version-details.ts

```diff
@@ -0,0 +1,362 @@
+import { Subscription, Asset, Build } from 'src/maestro-client/models';
+import { Observable, combineLatest, of } from 'rxjs';
+import { map, switchMap, shareReplay } from 'rxjs/operators';
+import { StatefulResult, statefulPipe } from 'src/stateful';
+import { BuildService } from 'src/app/services/build.service';
+import { WrappedError, Loading, statefulSwitchMap } from 'src/stateful/helpers';
+import { HttpHeaders, HttpClient } from '@angular/common/http';
+import { DependencyDetail } from './dependency-detail';
+
+export class VersionDetails {
+
+  // <DependencyName, DependencyDetails>
+  public allDependencies: Record<string, DependencyDetail> = {};
+  public dependenciesForEvaluation: Record<string, DependencyDetail> = {};
+
+  // <SubId, Subscription>
+  public unusedSubscriptions: Record<string, Subscription> = {};
+  public conflictingSubscriptions: Record<string, Subscription[]> = {};
+  public unableToRetrieveAssetsFor: Record<string, Subscription> = {};
+
+  // Dependency Name
+  public dependenciesWithNoSubscription: string[] = new Array();
+
+  // Subscription ID
+  public isToolsetSubscription: Record<string, boolean> = {};
+
+  constructor(inputFile: XMLDocument) {
+    const productElements = inputFile.getElementsByTagName("ProductDependencies");
+
+    if (productElements.length != 0) {
```

Comment:
if there is no ProductDependencies element (only toolset dependencies), does this fail?

Reply:
Same below: if there is no toolsetElement. Probably should have a null check.

Reply:
Ah, yes. I had logic handling it separately that was lost in the refactoring. Thanks for the catch!